packages/x-charts-pro/src/internals/plugins/useChartProZoom/useChartProZoom.ts

```diff
@@ -25,13 +25,32 @@ import {
   preventDefault,
   zoomAtPoint,
 } from './useChartProZoom.utils';
+
 // It is helpful to avoid the need to provide the possibly auto-generated id for each axis.
-function initializeZoomData(options: Record<AxisId, DefaultizedZoomOptions>) {
-  return Object.values(options).map(({ axisId, minStart: start, maxEnd: end }) => ({
-    axisId,
-    start,
-    end,
-  }));
+export function initializeZoomData(
+  options: Record<AxisId, DefaultizedZoomOptions>,
+  zoomData?: readonly ZoomData[],
+) {
+  const zoomDataMap = new Map<AxisId, ZoomData>();
+
+  zoomData?.forEach((zoom) => {
+    const option = options[zoom.axisId];
+    if (option) {
+      zoomDataMap.set(zoom.axisId, zoom);
+    }
+  });
+
+  return Object.values(options).map(({ axisId, minStart: start, maxEnd: end }) => {
+    if (zoomDataMap.has(axisId)) {
+      return zoomDataMap.get(axisId)!;
+    }
+
+    return {
+      axisId,
+      start,
+      end,
+    };
+  });
```

Comment:
We could possibly always merge the options. Though `ZoomData` requires the full object, so this is nitpick 🤔 

```suggestion
  return Object.values(options).map(({ axisId, minStart: start, maxEnd: end }) => {
    return {
      axisId,
      start,
      end,
      ...(zoomDataMap.get(axisId) ?? {})
    };
  });
```

Reply:
What would the benefit of this be? 🤔

Reply:
In case users passes a partial object

Reply:
Yeah, but why would they? I'm trying to think of a use case to justify implementing this, but I'm having a hard time